### DISCUSSION TITLE
How to trigger an Alert by creating play in playbook when someone execute sudo command

### DISCUSSION THREAD

**Discussion by sandraimmaculate** [Security-Onion-Solutions/securityonion#12319] [Unsupported Versions]
Hi, 
i created a play 
![Screenshot from 2024-02-07 13-19-24](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/908c31aa-7349-42f8-a73f-0a5ffbd6bce2)
![Screenshot from 2024-02-07 13-18-49](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/87cd2ed1-a4b4-4331-a3c9-d3e6b3c28495)

**I'm able to see about the play in kibana>discover dashboard when someone execute sudo command** 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/73223b22-c432-4edb-9771-5523deeed762)

**but not in security onion alert dashboard.**
do i have do anything else apart from this because im not getting alert still! 
please reply!

**Comment by InfosecGoon** [comment]
Does the query 

`process.command_line: *sudo*`

Return results in Hunt?

**Reply by sandraimmaculate** [reply]
process.command_line: *sudo*
no if i execute the above query in hunt dashboard I'm not getting any result! regarding that

**Reply by InfosecGoon** [reply]
Do you have an example event of someone running sudo in your environment? Sometimes it's easier to write the rules with an event record to work from.

**Reply by sandraimmaculate** [reply]
yea i have when i itself execute sudo command it will get store in /var/log/audit/audit.log  

cat audit.log.4 | grep "sudo"
type=USER_END msg=audit(17078118.56:148148): pid=186102 uid=102 auid=102 ses=21912 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 msg='op=PAM:session_close grantors=pam_keyinit,pam_limits,pam_systemd,pam_unix acct="root" exe="/usr/bin/sudo" hostname=? addr=? terminal=/dev/pts/0 res=success'UID="sandra" AUID="sandra"

here is an example event of someone running sudo in environment
even this log event, i can see in security onion because i added the system integration audit

**Reply by InfosecGoon** [reply]
Can you post a screenshot of this event in Security Onion? I want to see what the process.command_line field says.

**Reply by sandraimmaculate** [reply]
This is from security onion > kibana discover i filter it to log.file.path : /var/log/audit/audit.log and process.executable: "/usr/bin/sudo" 

![Screenshot from 2024-02-15 12-14-14](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/0dbdc39e-1768-424b-bc5e-dc10c2d0560e)

Sorry i have change the play now from process.command_line field  to 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/d765f880-2a40-4ab7-a99e-3df13db5f9d3)

Sorry idk how to filter this event in security onion dashboard or hunt because if i add the filter like process.executable: "/usr/bin/sudo" I'm not getting any result 


And i have a doubt that im able to see the sudo command log in kibana why not in security onion dashboard 

kindly help in this! thank you

**Reply by sandraimmaculate** [reply]
after changing the process.command_line to  process.executable: "/usr/bin/sudo"  i got the alert in alert dashboard BUT only on that time.
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/dd030261-3166-4555-9a62-7c9f37361a97)
on that day at 16/02/2024 at 6 pm I'm able to see the sudo command execution alert in alert dashboard, after that i have tried executing sudo command but i didn't got any alert inn alert dashboard. that was the first and last alert i got. why its happening like that!

but its not happening in kibana > discover im getting the info of sudo command execution when ever the sudo command is executed 
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/140578526/6a77ef44-23eb-4a0e-b91f-4668e8db3ea8)

**Reply by sandraimmaculate** [reply]
only sometimes its displaying the alert in alert dashboard about the sudo command execution. Sometimes it not displaying the alert even though sudo command is executed 

please suggest!
